### PR TITLE
API to check SPDX license

### PR DESCRIPTION
**Pull request by yugeshk**
Issue : #10 
This patch adds tests, models, serializers and API for implementation of the check license API.
Also updates README.md for API usage.
Signed-off: Yugesh Kothari yugesh.kothari@gmail.com

### CODE REVIEW

**Comment by yugeshk** [comment]
output - 
```
{
   "created" : "2018-03-18T03:28:28.505228Z",
   "status" : 201,
   "file" : "/media/apifiles/yugesh/AFL-1.1_i2bb6YP.txt",
   "result" : "The following license ID(s) match: AFL-1.1",
   "owner" : 2
}
```

**Comment by rtgdk** [comment]
Nice work! Looks good. Let me check it locally, if everything works I'll merge.

**Comment by rtgdk** [comment]
I have opened up some review points, please check and correct it. 
Also, this PR looks exactly similar to #18

**Comment by yugeshk** [comment]
@rtgdk Yes, i realized that too - I worked on this overnight and pushed as soon as it was working - didn't realize @susg had done so too. I based most of my code on the parts you asked to take inspiration from (following similar code style, structure and everything - the `models` were copy+paste from above :P ) - probably @susg did the same and thats why there are very little differences. :sweat_smile:

**Comment by yugeshk** [comment]
@rtgdk I can't find your reviews. Could you please guide?

**Comment by rtgdk** [comment]
You can the see the review points I made just above this chat. Or just go to the `Files Changed` tab and look for my comments there.

**Comment by yugeshk** [comment]
@rtgdk I'm sorry - but there are no comments at both places. Also, you're not showing up on the "Reviewers" section yet.

**Comment by susg** [comment]
Yeah, same thing for #18 .

**Comment by rtgdk** [comment]
I hope you can see them now.

**Comment by yugeshk** [comment]
Yes. Ive updated that.

**Comment by rtgdk** [comment]
Although your work is too similar to #18 , I'll merge the diff between this and the other one. Since he started working on this one first and the code is too similar I'll merge his first and then yours.

File: src/api/views.py [hunk 1]
```diff
@@ -408,4 +408,89 @@ def compare(request):
...
+                if request.FILES["file"]:
+                    """ Saving file to the media directory """
+                    myfile = request.FILES['file']
+                    licensetext = myfile.read()
+                    folder = "api/"+str(request.user) +"/"+ str(int(time()))
+                    fs = FileSystemStorage(location=settings.MEDIA_ROOT +"/"+ folder,
+                        base_url=urljoin(settings.MEDIA_URL, folder+'/')
+                        )
+                    filename = fs.save(myfile.name, myfile)
+                    uploaded_file_url = fs.url(filename)
```
Comment (techytushar) [inline src/api/views.py]: hey @yugeshk, the above lines of code will save the file twice in the media folder, once in the api folder and then in the apifiles folder, this was a bug which was fixed in #17, so please have a look at that code and try to modify this code accordingly, feel free to ask any doubts slightly_smiling_face

**Pull request closed without merging**